### PR TITLE
Ayomiku cu #860q6d3gu migrate the build process to generate food pages

### PR DESCRIPTION
**Pull request by Ayomikuolatunji**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,11 +95,11 @@
     "@types/react-slick": "^0.23.10",
     "babel-loader": "^8.3.0",
     "husky": "^8.0.0",
+    "postcss": "^8.4.21",
```
Comment (hhimanshu) [inline package.json:98]: Are we using this library? If yes, where?

File: README.md [hunk 2]
```diff
@@ -172,11 +172,11 @@ To run this project, you'll need to set up some environment variables. These var
    npm run build:env
    ```
```
Comment (hhimanshu) [inline README.md:173]: According to latest `package.json`, this should be `build:env:dev`

File: src/gatsby/createPages.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+// gatsby-node.js
```
Comment (hhimanshu) [inline src/gatsby/createPages.ts:1]: do we need this comment?

File: src/gatsby/siteMetaData.ts [hunk 4]
```diff
@@ -5,4 +5,5 @@ export const siteMetadata: SeoProps = {
   description: `Example project for the Gatsby Head API`,
   image: `/gatsby-icon.png`,
   siteUrl: `https://www.yourdomain.tld`,
+  slug: "",
```
Comment (hhimanshu) [inline src/gatsby/siteMetaData.ts:8]: Why are we adding Slug as SeoProps?

File: src/generators/DetailsPageGenerator.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        },
+      });
+      ffSearchIndex.push({
+        name: food.name,
+        category: food.category,
+        url: `/${pagePath}`,
+      });
+    }
+  });
+  // writeJsonToFile(`${indexFileName}.json`, ffSearchIndex);
```
Comment (hhimanshu) [inline src/generators/DetailsPageGenerator.ts:57]: We should enable index creation to ensure that index files are created correctly

File: src/helpers/utils.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
+// import fs from "fs";
```
Comment (hhimanshu) [inline src/helpers/utils.ts:1]: Remove unused comment

File: src/helpers/utils.ts [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+ */
+export const spaceToDashes = (name: string) => {
+  const pathname = name
+    .toLowerCase()
+    .trim()
+    .replace(/[^\w]+/g, "-");
+  return pathname.endsWith("-") ? pathname.substr(0, pathname.length - 1) : pathname;
+};
+
+// export const writeJsonToFile = (fileName: string, jsonData: Object[]) => {
```
Comment (hhimanshu) [inline src/helpers/utils.ts]: We should enable this code and write index

File: src/hooks/useSiteMetadata.ts [hunk 8]
```diff
@@ -10,11 +10,11 @@ const useSiteMetadata = () => {
           description
           image
           siteUrl
+          slug
```
Comment (hhimanshu) [inline src/hooks/useSiteMetadata.ts:13]: Is there a reason to include `slug`?

File: src/models/components.d.ts [hunk 9]
```diff
@@ -70,20 +71,16 @@ export interface compareTableItem {
...
-export interface NutritionTableItem {
+export interface NutritionTableRow {
   nutrient: string;
-  dailyValue: number | null;
-  amount: string | null;
+  dailyValue?: number | null;
+  amount?: number;
+  amountUnit?: string;
+  nutrientGroup: string;
   rdi: {
```
Comment (hhimanshu) [inline src/models/components.d.ts:80]: I think the entire `rdi` is optional (`?`). If it is present, then `value` and `weight` do exist. Also `weight` can be renamed as `servingUnitSize`

File: src/screens/DetailsPageScreens/DetailsPageScreen.tsx [hunk 10]
```diff
@@ -42,47 +39,59 @@ const DetailsPageScreen: React.FC<DetailsPageScreenProps> = ({
   const [_, setSelectedTitle] = useState("");
   const [selectedTabItem, setSelectedTabItem] = useState("Nutrition");
 
+  const onSelectItem = () => {
+    navigate("/");
```
Comment (hhimanshu) [inline src/screens/DetailsPageScreens/DetailsPageScreen.tsx:43]: AFAIK, we are currently not navigating users anywhere from the Detail page. Is there a reason to add `navigate()` here?

File: src/templates/DetailsPage.tsx [hunk 11]
```diff
@@ -0,0 +1,236 @@
...
+  const { food, recommendedDailyIntakes } = pageContext as any;
+  const [rows, setRows] = useState<any[]>([]);
+  const { selectedLifeStage, selectedAge, selectedNutrients, setSelectedAge } = useStore(
+    (state) => state
+  );
+  const thisFood = food as any;
+  const allRdis = recommendedDailyIntakes as any[];
+  const nutritionFacts: NutritionFact[] = generateRdiForFood(thisFood, allRdis);
+
+  const getSelectedNutrients = (nutrients: SelectedNutrient[]) => {
```
Comment (hhimanshu) [inline src/templates/DetailsPage.tsx:57]: This functions seems duplicate of `getSelectedNutrients` in `utils.ts`. Please remove one of them

File: package.json [hunk 12]
```diff
@@ -16,56 +16,65 @@
...
     "@mui/icons-material": "^5.10.15",
-    "@mui/material": "latest",
+    "@mui/material": "^5.11.14",
     "@mui/styles": "^5.10.15",
-    "@mui/x-data-grid": "^5.17.12",
     "@storybook/addon-viewport": "^6.5.16",
     "algoliasearch": "4.9.1",
     "classnames": "^2.3.2",
     "dotenv": "^16.0.3",
+    "fs": "^0.0.1-security",
```
[the remark this answers is not in the document]
Comment (Ayomikuolatunji) [reply package.json:47]: I wanted to use it but need now. Removed

File: package.json [hunk 13]
```diff
@@ -16,56 +16,65 @@
     "serve": "gatsby serve",
     "clean": "gatsby clean",
     "typecheck": "tsc --noEmit",
-    "storybook": "start-storybook -p 6006",
-    "build-storybook": "build-storybook",
+    "storybook": "npm run download-data && start-storybook -p 6006",
```
[the remark this answers is not in the document]
Comment (Ayomikuolatunji) [reply package.json]: Removed as well.

File: README.md (hunk 2 above)
Comment (Ayomikuolatunji) [reply README.md:173]: fixed

File: src/gatsby/createPages.ts (hunk 3 above)
Comment (Ayomikuolatunji) [reply src/gatsby/createPages.ts:1]: removed

File: src/gatsby/siteMetaData.ts (hunk 4 above)
Comment (Ayomikuolatunji) [reply src/gatsby/siteMetaData.ts:8]: removed

File: src/generators/DetailsPageGenerator.ts (hunk 5 above)
Comment (Ayomikuolatunji) [reply src/generators/DetailsPageGenerator.ts:57]: I saw the code in other forkfacts. Removed

File: src/helpers/utils.ts (hunk 7 above)
Comment (Ayomikuolatunji) [reply src/helpers/utils.ts]: removed. Not needed

File: src/helpers/utils.ts (hunk 6 above)
Comment (Ayomikuolatunji) [reply src/helpers/utils.ts:1]: fixed

File: src/hooks/useSiteMetadata.ts (hunk 8 above)
Comment (Ayomikuolatunji) [reply src/hooks/useSiteMetadata.ts:13]: Lol. Removed

File: src/models/components.d.ts (hunk 9 above)
Comment (Ayomikuolatunji) [reply src/models/components.d.ts:80]: udpated

File: src/screens/DetailsPageScreens/DetailsPageScreen.tsx (hunk 10 above)
Comment (Ayomikuolatunji) [reply src/screens/DetailsPageScreens/DetailsPageScreen.tsx:43]: This is for the go back button

File: static/temp/foods.json [hunk 14]
```diff
@@ -88,11 +88,11 @@
             "icon": "DirectionsCarFilledOutlinedIcon"
           },
           {
-            "name": "Male",
+            "name": "males",
```
[the remark this answers is not in the document]
Comment (Ayomikuolatunji) [reply static/temp/foods.json:91]: removed

File: src/templates/DetailsPage.tsx (hunk 11 above)
Comment (Ayomikuolatunji) [reply src/templates/DetailsPage.tsx:57]: No I am not sure of that. I just confirmed that

File: package.json (hunk 1 above)
Comment (Ayomikuolatunji) [reply package.json:98]: Yes it was installed with the gatsby boilerplate.

**Review by hhimanshu — APPROVED** [review]

**Pull request merged**